packages/node_modules/@node-red/editor-client/src/js/ui/tour/tourGuide.js

```diff
@@ -199,6 +258,51 @@ RED.tourGuide = (function() {
         return property[currentLang]||property['en-US']||property[availableLangs[0]]
 
     }
+
+    /**
+     * @typedef {"right"|"left"|"bottom"|"top"|"inset"} Direction A valid CSS
+     * direction. The list is not exhaustive.
+     *
+     * @typedef {object} TourStepState
+     * @property {number} index The index of the current step
+     * @property {number} count The total number of steps
+     * 
+     * @typedef {object} Wait
+     * @property {"dom-event"|"nr-event"} type The emitter type. If `dom-event`
+     * use {@link Wait.element element} otherwise use {@link Wait.filter filter}
+     * @property {string} event The event name
+     * @property {string|((this: TourStepState) => JQuery)} element The DOM
+     * element to attach the event listener
+     * @property {(event: object) => boolean} filter An function called when
+     * the event is triggered to filter the target action.
+     *
+     * @typedef {object} TourStep Represents a step of the Tour Guide
+     * @property {(this: TourStepState, done?: () => void) => void} [complete]
+     * A function called when the dialog is being closed
+     * @property {string} [description] The description of the dialog
+     * @property {Direction} [direction = "bottom"] The direction of the dialog
+     * relative to the {@link TourStep.element element}. Default `bottom`.
+     * @property {string|JQuery|((this: TourStepState) => JQuery)} [element]
+     * The DOM element to highlight.
+     * @property {Direction} [fallback] On mouse hover, stop highlighting the 
+     * {@link TourStep.element element} and move the dialog in another direction
+     * @property {string} [image] The path to the image to be integrated into
+     * the dialog
+     * @property {boolean} [interactive] Either to place the dialog on top of
+     * the overlapping stack. Enabled if {@link TourStep.wait wait} is used
+     * @property {(this: TourStepState, done?: () => void) => void} [prepare]
+     * A function called when the dialog is being built
+     * @property {string} [title] The title of the dialog
+     * @property {string} [titleIcon] The icon to add to the title in the dialog
+     * @property {Wait} [wait] Wait for an event to be triggered before closing
+     * the dialog
+     * @property {string|number} [width] The width of the dialog
+     *
+     * @param {TourStep} step A step of the Tour Guide
+     * @param {TourStepState} state The state of a the Tour Guide step
+     * @param {() => void} done A function called when the dialog is closed.
+     * After the {@link TourStep.complete complete} event
+     */
     function runTourStep(step, state, done) {
```

Comment:
Not sure this is the right place to be documenting the tour json format - this is an internal function, so the generated doc isn't going to get surfaced anywhere useful.  I'm also concerned this will get out of sync with the implementation.